app/write_square_test.go

```diff
@@ -75,9 +75,22 @@ func Test_finalizeLayout(t *testing.T) {
 			nonreserveStart: 32,
 			ptxs: generateParsedTxsWithNIDs(
 				[][]byte{ns1, ns1, ns1},
-				[]int{1000, 10000, 100000},
+				[]int{1000, 10000, 100000}, // blob share lengths of 2, 20, 199 respectively
 			),
-			expectedIndexes: []uint32{32, 48, 128},
+			expectedIndexes: []uint32{
+				// BlobMinSquareSize(2) = 2 so the first blob has to start at the
+				// next multiple of 2 > 32 which is 32. This blob occupies
+				// shares 32 to 34.
+				32,
+				// BlobMinSquareSize(20) = 8 so the second blob has to start at
+				// the next multiple of 8 > 34 which is 40. This blob occupies
+				// shares 40 to 48.
+				40,
+				// BlobMinSquareSize(199) = 16 so the third blob has to start at
+				// the next multiple of 16 > 48 which is 64. This blob occupies
+				// shares 64 to 263.
+				64,
+			},
```

Comment:
[non-blocking][nit]
```suggestion
			expectedIndexes: []uint32{
				// BlobMinSquareSize(2) = 2 so the first blob has to start at the
				// next multiple of 2 > 32 which is 32. This blob occupies
				// shares 32 to 33.
				32,
				// BlobMinSquareSize(20) = 8 so the second blob has to start at
				// the next multiple of 8 > 34 which is 40. This blob occupies
				// shares 40 to 59
				40,
				// BlobMinSquareSize(199) = 16 so the third blob has to start at
				// the next multiple of 16 > 48 which is 64. This blob occupies
				// shares 64 to 262.
				64,
			},
```
If the blob occupies shares 2 to 4 it would occupy 3 shares in my understanding. Also depends on how you start counting 0 or 1. I started counting at 0 for this example. The last share of the square would be 263 in this case. The last square of the blob would be 262.

Reply:
Agreed, thanks for pointing this out! This example assumes 0-based indexing. The last index in the ranges are all off by one so I'll update to your suggestion in a follow-up PR.

Reply:
heads up the second one is off by 11

Reply:
Agreed, it wasn't just off by one. Fixed in https://github.com/celestiaorg/celestia-app/pull/1185